docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
+  * [Quote Management](#42-feature-quote-management)
+  * [Progress Tracker](#43-progress-tracker)
+  * [Category Management](#44-feature-category-management)
+  * [Rating System for books](#45-feature-rating-system-for-books)
+- [Appendix: Requirements](#appendix-requirements)
+  * [Product Scope](#product-scope)
+  * [User Stories](#user-stories)
+  * [Non-Functional Requirments](#non-functional-requirements)
+  * [Glossary](#glossary)
+- [Appendix: Instructions for manual testing](#appendix-instructions-for-manual-testing)
+  * [Launch and Shutdown](#launch-and-shutdown)
+  * [Adding a book](#adding-a-book)
+  * [List all existing books](#list-all-existing-books)
+  * [List book details](#list-book-details)
+  * [List books by author](#list-books-by-author)
+  * [Find books by keyword](#find-books-by-keyword)
+  * [Delete books](#delete-books)
+  * [Edit book](#edit-book)
+  * [Adding a quote](#adding-a-quote)
+  * [Listing all quotes](#listing-all-quotes)
+  * [Listing quotes by a specific author](#listing-quotes-by-a-specific-author)
+  * [Listing quotes from a specific reference](#listing-quotes-from-a-specific-reference)
+  * [Listing quotes from a specific reference and by a specific author](#listing-quotes-from-a-specific-reference-and-by-a-specific-author)
+  * [Editing a quote](#editing-a-quote)
+  * [Deleting a quote](#deleting-a-quote)
+  * [Finding a quote](#finding-a-quote)
+  * [Adding reflection to quote](#adding-reflection-to-a-quote)
+  * [Listing reflection of a quote](#listing-reflection-of-a-quote)
+  * [Editing reflection of a quote](#editing-reflection-of-a-quote)
+  * [Deleting reflection of a quote](#deleting-reflection-of-a-quote)
+  * [Adding a bookmark to book](#adding-a-bookmark-to-book)
+  * [List all existing bookmarks](#list-all-existing-bookmarks)
+  * [Deleting an existing bookmark](#deleting-an-existing-bookmark)
+  * [Editing an existing bookmark](#edit-an-existing-bookmark)
+  * [Adding a task to todo list](#adding-a-task-to-todo-list)
+  * [Listing all existing tasks](#listing-all-existing-tasks)
+  * [Marking an existing task as done](#marking-an-existing-task-as-done)
+  * [Deleting an existing task](#deleting-an-existing-task)
+  * [Adding categories](#adding-categories)
+  * [Listing all categories](#listing-all-categories)
+  * [Listing a specific category](#listing-a-specific-category)
+  * [Deleting existing categories](#deleting-existing-categories)
+  * [Editing an existing category](#editing-an-existing-category)
+  * [Finding an existing category](#finding-an-existing-category)
+  * [Adding a book rating](#adding-a-book-rating)
+  * [Listing all existing book ratings](#listing-all-existing-book-ratings)
+  * [Listing books of a specific book rating](#listing-books-of-a-specific-book-rating)
+  * [Deleting a book rating](#deleting-a-book-rating)
+  * [Editing a book rating](#editing-a-book-rating)
+  * [Finding a book rating](#finding-a-book-rating)
```

Comment:
Does this look quite apprehensive for the new developer who's just came up with the project. Can you nest these in a drop-down list just like in the module website?